# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsChild.h
# Commit: 13e00068cb2b
# Full Hash: 13e00068cb2bf553f6badfbacbbc7decfcfdf4bf
# Author: Birunthan Mohanathas <birunthan@mohanathas.com>
# Date: 2015-08-23 21:18:49
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Part 5: Send two records with every ObjectStoreCursorResponse. r=khuey
# ==============================================================================

diff -r 8e9fb56cc22a -r 13e00068cb2b dom/indexedDB/ActorsChild.h
--- a/dom/indexedDB/ActorsChild.h	Wed Aug 19 14:59:28 2015 -0700
+++ b/dom/indexedDB/ActorsChild.h	Wed Aug 19 14:59:29 2015 -0700
@@ -626,6 +626,17 @@
 
   class DelayedActionRunnable;
 
+  struct CachedResponse
+  {
+    CachedResponse();
+
+    explicit CachedResponse(CachedResponse&& aOther);
+
+    Key mKey;
+    Key mObjectKey;
+    StructuredCloneReadInfo mCloneInfo;
+  };
+
   IDBRequest* mRequest;
   IDBTransaction* mTransaction;
   IDBObjectStore* mObjectStore;
@@ -642,6 +653,8 @@
   PRThread* mOwningThread;
 #endif
 
+  nsTArray<CachedResponse> mCachedResponses;
+
 public:
   BackgroundCursorChild(IDBRequest* aRequest,
                         IDBObjectStore* aObjectStore,
@@ -665,6 +678,9 @@
   void
   SendDeleteMeInternal();
 
+  void
+  InvalidateCachedResponses();
+
   IDBRequest*
   GetRequest() const
   {
@@ -703,6 +719,9 @@
   ~BackgroundCursorChild();
 
   void
+  SendDelayedContinueInternal();
+
+  void
   HandleResponse(nsresult aResponse);
 
   void